extensions/commands/diff/cmd_check_diff.py

```diff
@@ -0,0 +1,176 @@
+import json
+import subprocess
+import os
+
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+
+def output_json(msg):
+    return json.dumps({"greet": msg})
+
+
+@conan_command(group="utils", formatters={"json": output_json})
+def check_diff(conan_api: ConanAPI, parser, *args):
+    """
+    Command to get the diff between versions
+    """
+    parser.add_argument("path", help="Path to the conanfile")
+    parser.add_argument("v1", help="Old version")
+    parser.add_argument("v2", help="New version")
+    parser.add_argument("-s", "--split_diff", action="store_true")
+    args = parser.parse_args(*args)
+
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=True)
+    enabled_remotes = conan_api.remotes.list()
+
+    src_dir = os.path.join(path, "src") #TODO can break if the src_folder is not defined as "src"
```

Comment:
I know we talked about using the sources if they are available from the Conan cache, but, if they are not:

Make this 2 temp folders, and call `source()` on each of them, then diff both folders directly using the `diff` command or whatever python library is available.

This will let you skip creating a git repo and the flow won't break if the source folder is not named src